<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Sailalitha Bollepalli" />


<title>EpiSmokEr - Epigenetic Smoking status Estimator- R package</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">EpiSmokEr - Epigenetic Smoking status Estimator- R package</h1>
<h4 class="author"><em>Sailalitha Bollepalli</em></h4>
<h4 class="date"><em>Last updated on: 2018-09-01</em></h4>


<div id="TOC">
<ul>
<li><a href="#introduction">Introduction</a><ul>
<li><a href="#four-options-in-epismoker">Four options in EpiSmokEr</a></li>
<li><a href="#features-of-epismoker">Features of <strong>EpiSmokEr</strong></a></li>
</ul></li>
<li><a href="#installation">Installation</a></li>
<li><a href="#input-data">Input Data</a></li>
<li><a href="#smoking-score-calculation-and-predicton-of-smoking-status">Smoking Score calculation and predicton of smoking status</a><ul>
<li><a href="#a.-from-idat-files">A. From idat files</a><ul>
<li><a href="#please-note">Please note:</a></li>
<li><a href="#smoking-status-sst-prediction">1. Smoking Status (SSt) Prediction</a></li>
<li><a href="#smoking-score-ssc">2. Smoking Score (SSc)</a></li>
<li><a href="#methylation-score-ms">3. Methylation Score (MS)</a></li>
<li><a href="#compreshensive-approach-all">4. Compreshensive approach (all)</a></li>
</ul></li>
<li><a href="#b.-from-methylation-matrix">B. From methylation matrix</a></li>
</ul></li>
<li><a href="#generate-results">Generate Results</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>Introduction</h1>
<hr />
<p>Several independent studies have reported the high impact of smoking on DNA methylation with current, never and former smokers exhibiting different methylation profiles. Recently, usage of DNA methylation as a proxy measure to estimate smoking status has been proposed <span class="citation">(Elliott et al. 2014)</span>. <strong>EpiSmokEr</strong> serves as a comprehensive package with four options to estimate smoking status based on methylation data.</p>
<p>This vignette gives an overview of how to use the <strong>EpiSmokEr</strong> R package.</p>
<div id="four-options-in-epismoker" class="section level2">
<h2>Four options in EpiSmokEr</h2>
<p><strong>EpiSmokEr</strong> provides four options to calculate smoking score and predicts smoking status from whole blood Infinium HumanMethylation450 data.</p>
<ol style="list-style-type: decimal">
<li><strong>Smoking Status (SSt)</strong> provides smoking probabilities for each individual and <strong>predicts smoking status</strong> of the individual. This approach uses weights of 121 CpGs selected by multinomial LASSO approach for each class of smoking status i.e current, former and never smokers. In addition to these 121 CpGs, sex and intercept terms are also used in the calculation. These coefficients used by SSt approach can be viewed as follows:</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;SSt_coefficients.rda&quot;</span>, <span class="dt">package=</span> <span class="st">&quot;EpiSmokEr&quot;</span>))
knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">head</span>(SSt_coefficients, <span class="dt">caption =</span> <span class="st">&quot;121 CpGs and coefficients selected by Multinomial LASSO approach&quot;</span>))</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">CS_final_coefs</th>
<th align="right">FS_final_coefs</th>
<th align="right">NS_final_coefs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>intercept</td>
<td align="right">11.3824302</td>
<td align="right">-1.3625468</td>
<td align="right">-10.0198834</td>
</tr>
<tr class="even">
<td>sexM</td>
<td align="right">-0.3003657</td>
<td align="right">0.4071201</td>
<td align="right">-0.1067544</td>
</tr>
<tr class="odd">
<td>cg00006626</td>
<td align="right">-1.7139848</td>
<td align="right">1.3992256</td>
<td align="right">0.3147592</td>
</tr>
<tr class="even">
<td>cg03133799</td>
<td align="right">0.0960137</td>
<td align="right">0.0923306</td>
<td align="right">-0.1883443</td>
</tr>
<tr class="odd">
<td>cg06442199</td>
<td align="right">-0.0007457</td>
<td align="right">0.0061609</td>
<td align="right">-0.0054152</td>
</tr>
<tr class="even">
<td>cg10006428</td>
<td align="right">-0.5519797</td>
<td align="right">0.8864383</td>
<td align="right">-0.3344586</td>
</tr>
</tbody>
</table>
<ol start="2" style="list-style-type: decimal">
<li><p><strong>Smoking Score (SSc)</strong> is calculated based on the approach outlined in <span class="citation">(Elliott et al. 2014)</span>. Elliott <em>et al</em> computed smoking scores using 187 smoking-associated CpGs identified in an EWAS comparing current Vs never smokers <span class="citation">(Zeilinger et al. 2013)</span>. For a detailed explanation of smoking score calculation please refer to <span class="citation">(Elliott et al. 2014)</span>. Briefly, weights and reference beta values are obtained from Zellinger <em>et al</em>. The 187 CpG sites are divided into two groups based on their methylation levels in never smokers. For CpGs with decreased methylation, the cohort beta values are subtracted from the reference beta values. For CpGs with increased methylation reference beta values are subtracted from cohort beta values. Then beta values are multiplied by the corresponding weights. A smoking score is then obtained by summation across all the 187 CpG site scores.</p></li>
<li><p><strong>Methylation Score (MS)</strong> approach was proposed by Zhang <em>et al </em> to distinguish current smokers from never smokers, and former smokers from never smokers using a stepwise selection of logistic regression models <span class="citation">(Zhang et al. 2016)</span>. Weights of 4 CpGs provided by Zhang <em>et al</em> were used to obtain methylation scores.</p></li>
<li><p><strong>all</strong> a comprehensive approach which provides results from all the three methods described above.</p></li>
</ol>
</div>
<div id="features-of-epismoker" class="section level2">
<h2>Features of <strong>EpiSmokEr</strong></h2>
<ul>
<li>Can perform smoking score calculation starting from idat files.</li>
<li>Normalization of datasets using channel and colour specific quantile normalization.</li>
<li>Estimates Smoking Status and can provide Smoking Score and Methylation Scores.</li>
<li>Prediction of smoking status based on the whole blood methylation data.</li>
<li>Results provided in html and csv format.</li>
</ul>
</div>
</div>
<div id="installation" class="section level1">
<h1>Installation</h1>
<hr />
<p>To install EpiSmokEr, start R and then type the following commands:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&quot;http://bioconductor.org/biocLite.R&quot;</span>)
<span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>) <span class="co"># if you don't have the package, run install.packages(&quot;devtools&quot;)</span>
<span class="kw">library</span>(devtools)
<span class="kw">install_github</span>(<span class="st">&quot;sailalithabollepalli/EpiSmokEr&quot;</span>)</code></pre></div>
<p><strong>EpiSmokEr</strong> depends on the following packages:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(IlluminaHumanMethylation450kmanifest)
<span class="kw">library</span>(IlluminaHumanMethylation450kanno.ilmn12.hg19)
<span class="kw">library</span>(minfi)
<span class="kw">library</span>(htmlTable)
<span class="kw">library</span>(rmarkdown)</code></pre></div>
<p>After installing <strong>EpiSmokEr</strong>, load the package along with its dependencies:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
<span class="kw">library</span>(EpiSmokEr)  
<span class="kw">library</span>(IlluminaHumanMethylation450kmanifest)
<span class="kw">library</span>(IlluminaHumanMethylation450kanno.ilmn12.hg19)
<span class="kw">library</span>(minfi)
<span class="kw">library</span>(htmlTable)
<span class="kw">library</span>(rmarkdown)
})</code></pre></div>
</div>
<div id="input-data" class="section level1">
<h1>Input Data</h1>
<hr />
<p>To estimate smoking status and scores from the whole blood methylaion, EpiSmokEr simply requires methylation data as an input. Input data could be either raw methylation data in the form of idat files or a normalised methylation matrix in the beta scale ranging in between 0 and 1. For SSt method, a sample sheet with <strong>sex</strong> status is needed to complement the methylation data.</p>
<p>To demonstrate the working of the EpiSmokEr we chose a peripheral blood leukocytes dataset from GEO <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=gse42861">GSE42861</a><span class="citation">(Liu et al. 2013)</span>. We use only 6 samples from this dataset to minimise the package size and running time.</p>
<div id="following-data-objects-are-provided-along-with-the-package-and-are-used-in-the-examples-below." class="section level4">
<h4>Following data objects are provided along with the package and are used in the examples below.</h4>
<ul>
<li>Raw data: idat files of 6 subjects from GSE42861.</li>
<li>Methylation matrix: A subset of 1000 CpG probes from the normalised methylation matrix (which contained all the CpGs required by EpiSmokEr).</li>
<li>A sample sheet with gender information</li>
</ul>
</div>
</div>
<div id="smoking-score-calculation-and-predicton-of-smoking-status" class="section level1">
<h1>Smoking Score calculation and predicton of smoking status</h1>
<hr />
<p>We will now show 2 ways of providing input data to predict smoking status:</p>
<div id="a.-from-idat-files" class="section level2">
<h2>A. From idat files</h2>
<p>Here input data are in the form of idat files. <code>minfi</code><span class="citation">(Aryee et al. 2014)</span> package is used to read the idat files. <strong>normaliseData</strong> function has a suite of customised internal functions to normalise and calculate beta values from the idat files. Color and channel specific normalisation was performed on the training dataset used in the SSt approach. Quantiles from this training dataset are used to adjust the distribution of your input data comparable to the training dataset. Following the normalisation, beta values are calculated.</p>
<div id="please-note" class="section level3">
<h3>Please note:</h3>
<ul>
<li>Sample sheet must be included in the same folder as the idat files.</li>
<li>Rownames of samplesheet must match the names of the samples i.e idat files, to facilitate matching of phenotype data with the corresponding methylation data.</li>
<li>For SSt method, sample sheet should include a column with gender information marked as <strong>sex</strong>, in the format of 1 and 2 representing men and women respectively.</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rawdata &lt;-<span class="st"> </span><span class="kw">loadData</span>(<span class="dt">idatPath =</span> <span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;EpiSmokEr&quot;</span>))</code></pre></div>
<pre><code>## Loading idat files.</code></pre>
<pre><code>## [read.metharray.sheet] Found the following CSV files:</code></pre>
<pre><code>## [1] &quot;C:/Users/sairam/Documents/R/win-library/3.5/EpiSmokEr/extdata/samplesheet_GSE42861.csv&quot;</code></pre>
<pre><code>## Dataset has 6 samples.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samplesheet &lt;-<span class="kw">read.csv</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;samplesheet_GSE42861.csv&quot;</span>, <span class="dt">package=</span><span class="st">&quot;EpiSmokEr&quot;</span>), <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">sep=</span><span class="st">&quot;,&quot;</span>)
knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">head</span>(samplesheet, <span class="dv">5</span>))</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Age</th>
<th align="left">Gender</th>
<th align="left">Smokingstatus.selfReported.</th>
<th align="right">sex</th>
<th align="right">Slide</th>
<th align="left">Array</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GSM1051533_7800246024_R03C02</td>
<td align="right">55</td>
<td align="left">m</td>
<td align="left">current</td>
<td align="right">1</td>
<td align="right">7800246024</td>
<td align="left">R03C02</td>
</tr>
<tr class="even">
<td>GSM1051534_7800246024_R04C02</td>
<td align="right">37</td>
<td align="left">m</td>
<td align="left">never</td>
<td align="right">1</td>
<td align="right">7800246024</td>
<td align="left">R04C02</td>
</tr>
<tr class="odd">
<td>GSM1051538_7800246085_R02C01</td>
<td align="right">47</td>
<td align="left">m</td>
<td align="left">ex</td>
<td align="right">1</td>
<td align="right">7800246085</td>
<td align="left">R02C01</td>
</tr>
<tr class="even">
<td>GSM1051539_7800246085_R03C01</td>
<td align="right">69</td>
<td align="left">m</td>
<td align="left">never</td>
<td align="right">1</td>
<td align="right">7800246085</td>
<td align="left">R03C01</td>
</tr>
<tr class="odd">
<td>GSM1051584_7800246137_R01C01</td>
<td align="right">62</td>
<td align="left">f</td>
<td align="left">ex</td>
<td align="right">2</td>
<td align="right">7800246137</td>
<td align="left">R01C01</td>
</tr>
</tbody>
</table>
<p>When raw methylation data is available (idat files) then it is highly recommended to use the same normalisation method implemented in the original article which proposed that specific smoking-score method. All the three smoking score methods (SSt, MS and SSc) we have considered here uses 3 different normalisation methods (QN, SQN and ILM). Therefore we provide three normalisation options to normalise the datasets accordingly and then calculate the smoking score following that specific method.</p>
<p>Quantiles saved from color, channel and probe based quantile normalisation implemented on the training dataset will be used to normalise data for SSt.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dataset_QN &lt;-<span class="st"> </span><span class="kw">normalizeData</span>(<span class="dt">RGset=</span>rawdata, <span class="dt">normMethod =</span> <span class="st">&quot;QN&quot;</span>) </code></pre></div>
<pre><code>## Loading RGset object.</code></pre>
<pre><code>## Dataset has 6 samples.</code></pre>
<p>To use SSc, dataset is normalised using subset quantile normalization <span class="citation">(Touleimat and Tost 2012)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dataset_SQN &lt;-<span class="st"> </span><span class="kw">normalizeData</span>(<span class="dt">RGset=</span>rawdata, <span class="dt">normMethod =</span> <span class="st">&quot;SQN&quot;</span>)</code></pre></div>
<pre><code>## Loading RGset object.</code></pre>
<pre><code>## Dataset has 6 samples.</code></pre>
<pre><code>## Performing Subset Quantile Normalisation.</code></pre>
<pre><code>## [preprocessQuantile] Mapping to genome.</code></pre>
<pre><code>## [preprocessQuantile] Fixing outliers.</code></pre>
<pre><code>## [preprocessQuantile] Quantile normalizing.</code></pre>
<p>To use MS, data were normalized to internal controls provided by Illumina (Illumina normalization) and no additional background correction was done.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dataset_ILM &lt;-<span class="st"> </span><span class="kw">normalizeData</span>(<span class="dt">RGset=</span>rawdata, <span class="dt">normMethod =</span> <span class="st">&quot;ILM&quot;</span>)</code></pre></div>
<pre><code>## Loading RGset object.</code></pre>
<pre><code>## Dataset has 6 samples.</code></pre>
<pre><code>## Performing Illumina Normalisation.</code></pre>
<p>All the three normalisation methods can be performed using option “ALL”. However, the result <strong>dataset_ALL</strong> is saved as a list.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dataset_ALL &lt;-<span class="st"> </span><span class="kw">normalizeData</span>(<span class="dt">RGset=</span>rawdata, <span class="dt">normMethod =</span> <span class="st">&quot;ALL&quot;</span>)</code></pre></div>
<pre><code>## Loading RGset object.</code></pre>
<pre><code>## Dataset has 6 samples.</code></pre>
<pre><code>## Performing Subset Quantile Normalisation.</code></pre>
<pre><code>## [preprocessQuantile] Mapping to genome.</code></pre>
<pre><code>## [preprocessQuantile] Fixing outliers.</code></pre>
<pre><code>## [preprocessQuantile] Quantile normalizing.</code></pre>
<pre><code>## Performing Illumina Normalisation.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(dataset_ALL)</code></pre></div>
<pre><code>## List of 3
##  $ dataset_QN : num [1:452088, 1:6] 0.206 0.469 0.811 0.859 0.886 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:452088] &quot;cg00000165&quot; &quot;cg00000363&quot; &quot;cg00001364&quot; &quot;cg00001446&quot; ...
##   .. ..$ : chr [1:6] &quot;GSM1051533_7800246024_R03C02&quot; &quot;GSM1051534_7800246024_R04C02&quot; &quot;GSM1051538_7800246085_R02C01&quot; &quot;GSM1051539_7800246085_R03C01&quot; ...
##  $ dataset_ILM: num [1:485512, 1:6] 0.8349 0.304 0.8729 0.0317 0.8256 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:485512] &quot;cg00050873&quot; &quot;cg00212031&quot; &quot;cg00213748&quot; &quot;cg00214611&quot; ...
##   .. ..$ : chr [1:6] &quot;GSM1051533_7800246024_R03C02&quot; &quot;GSM1051534_7800246024_R04C02&quot; &quot;GSM1051538_7800246085_R02C01&quot; &quot;GSM1051539_7800246085_R03C01&quot; ...
##  $ dataset_SQN: num [1:485512, 1:6] 0.772 0.471 0.109 0.265 0.132 ...
##   ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. ..$ : chr [1:485512] &quot;cg13869341&quot; &quot;cg14008030&quot; &quot;cg12045430&quot; &quot;cg20826792&quot; ...
##   .. ..$ : chr [1:6] &quot;GSM1051533_7800246024_R03C02&quot; &quot;GSM1051534_7800246024_R04C02&quot; &quot;GSM1051538_7800246085_R02C01&quot; &quot;GSM1051539_7800246085_R03C01&quot; ...</code></pre>
<p>Individual normalised datasets can be obtained from <strong>dataset_ALL</strong> as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dataset_QN &lt;-<span class="st"> </span>dataset_ALL<span class="op">$</span>dataset_QN
dataset_ILM &lt;-<span class="st"> </span>dataset_ALL<span class="op">$</span>dataset_ILM
dataset_SQN &lt;-<span class="st"> </span>dataset_ALL<span class="op">$</span>dataset_SQN</code></pre></div>
<p>Now the <strong>dataset</strong> object can be passed on to the <strong>epismoker</strong> function to calculate smoking score and smoking status using four options available.</p>
<p>Once we have normalised the methylation data, we can then proceed with smoking status estimation. EpiSmokEr provides four options:</p>
</div>
<div id="smoking-status-sst-prediction" class="section level3">
<h3>1. Smoking Status (SSt) Prediction</h3>
<p>We derive the smoking probabilities, using the weights from 121 CpGs selected by Multinomial LASSO approach, along with the sex and intercept terms. We get smoking probabilities for three smoking statuses, i.e current, former and never smokers. Predicted smoking status of an individual is the class having the highest probability.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_SSt &lt;-<span class="st"> </span><span class="kw">epismoker</span>(<span class="dt">dataset=</span>dataset_QN, <span class="dt">samplesheet =</span> samplesheet, <span class="dt">method =</span> <span class="st">&quot;SSt&quot;</span>)</code></pre></div>
<pre><code>## ============================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Predicting Smoking Statuses &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ============================================</code></pre>
<pre><code>## Dataset has 121 of 121 CpGs required for smoking status estimation.
## Dataset has 121 of 121 CpGs required for smoking status estimation.
## Dataset has 121 of 121 CpGs required for smoking status estimation.</code></pre>
<pre><code>## ============================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Prediction of Smoking Statuses Completed &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ============================================</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_SSt</code></pre></div>
<pre><code>##                                                SampleName logOdds_CS
## GSM1051533_7800246024_R03C02 GSM1051533_7800246024_R03C02  1.2702463
## GSM1051534_7800246024_R04C02 GSM1051534_7800246024_R04C02 -1.2369648
## GSM1051538_7800246085_R02C01 GSM1051538_7800246085_R02C01 -0.7931685
## GSM1051539_7800246085_R03C01 GSM1051539_7800246085_R03C01 -1.2175070
## GSM1051584_7800246137_R01C01 GSM1051584_7800246137_R01C01 -0.7364596
## GSM1051585_7800246137_R02C01 GSM1051585_7800246137_R02C01 -1.5291249
##                              logOdds_FS logOdds_NS   probs_CS  probs_FS
## GSM1051533_7800246024_R03C02  0.7726038 -2.0428501 0.60813729 0.3697245
## GSM1051534_7800246024_R04C02  0.2631190  0.9738458 0.06846889 0.3068820
## GSM1051538_7800246085_R02C01  0.3565658  0.4366027 0.13196440 0.4166583
## GSM1051539_7800246085_R03C01  0.4712640  0.7462431 0.07386166 0.3998002
## GSM1051584_7800246137_R01C01  0.5094571  0.2270025 0.14090711 0.4898100
## GSM1051585_7800246137_R02C01  0.6709051  0.8582198 0.04782438 0.4316286
##                                probs_NS PredictedSmokingStatus
## GSM1051533_7800246024_R03C02 0.02213819         Current Smoker
## GSM1051534_7800246024_R04C02 0.62464911           Never Smoker
## GSM1051538_7800246085_R02C01 0.45137726           Never Smoker
## GSM1051539_7800246085_R03C01 0.52633818           Never Smoker
## GSM1051584_7800246137_R01C01 0.36928289          Former Smoker
## GSM1051585_7800246137_R02C01 0.52054702           Never Smoker</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># PredictedSmokingStatus represents the smoking status class assigned to the individual based on the calculated smoking probabilities.</span></code></pre></div>
</div>
<div id="smoking-score-ssc" class="section level3">
<h3>2. Smoking Score (SSc)</h3>
<p>In this method 187 CpGs are used to calculate smoking score as expained in <span class="citation">(Elliott et al. 2014)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_SSc &lt;-<span class="st"> </span><span class="kw">epismoker</span>(<span class="dt">dataset =</span> dataset_SQN, <span class="dt">method =</span> <span class="st">&quot;SSc&quot;</span>)</code></pre></div>
<pre><code>## =================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Smoking Score Calculation Started &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## =================================</code></pre>
<pre><code>## Dataset has 187 of 187 CpGs required for smoking status estimation.</code></pre>
<pre><code>## ====================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Smoking Score Calculation Completed &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ====================================</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_SSc</code></pre></div>
<pre><code>##                                                SampleName smokingScore
## GSM1051533_7800246024_R03C02 GSM1051533_7800246024_R03C02    7.5861170
## GSM1051534_7800246024_R04C02 GSM1051534_7800246024_R04C02    1.1805043
## GSM1051538_7800246085_R02C01 GSM1051538_7800246085_R02C01    1.1677897
## GSM1051539_7800246085_R03C01 GSM1051539_7800246085_R03C01   -2.0578117
## GSM1051584_7800246137_R01C01 GSM1051584_7800246137_R01C01   -0.2134197
## GSM1051585_7800246137_R02C01 GSM1051585_7800246137_R02C01   -4.7081811</code></pre>
</div>
<div id="methylation-score-ms" class="section level3">
<h3>3. Methylation Score (MS)</h3>
<p>In this method 4 CpGs are used to calculate methylation score as expained in <span class="citation">(Zhang et al. 2016)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_MS &lt;-<span class="st"> </span><span class="kw">epismoker</span>(<span class="dt">dataset =</span> dataset_ILM,  <span class="dt">method =</span> <span class="st">&quot;MS&quot;</span>)</code></pre></div>
<pre><code>## ================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Methylation Score Calculation Started &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## =================================</code></pre>
<pre><code>## Dataset has 4 of 4 CpGs required for Zhang et al method</code></pre>
<pre><code>## ==================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Methylation Score Calculation Completed &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ==================================</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_MS</code></pre></div>
<pre><code>##                                                SampleName methylationScore
## GSM1051533_7800246024_R03C02 GSM1051533_7800246024_R03C02        -4.284498
## GSM1051534_7800246024_R04C02 GSM1051534_7800246024_R04C02        -8.993818
## GSM1051538_7800246085_R02C01 GSM1051538_7800246085_R02C01        -6.732453
## GSM1051539_7800246085_R03C01 GSM1051539_7800246085_R03C01        -7.180410
## GSM1051584_7800246137_R01C01 GSM1051584_7800246137_R01C01        -7.068201
## GSM1051585_7800246137_R02C01 GSM1051585_7800246137_R02C01        -8.731180</code></pre>
</div>
<div id="compreshensive-approach-all" class="section level3">
<h3>4. Compreshensive approach (all)</h3>
<p>All three methods are run in this option and results are integrated into a single result object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_All &lt;-<span class="st"> </span><span class="kw">epismoker</span>(<span class="dt">dataset_QN=</span>dataset_QN, <span class="dt">dataset_ILM=</span>dataset_ILM, <span class="dt">dataset_SQN=</span>dataset_SQN, <span class="dt">samplesheet =</span> samplesheet, <span class="dt">method =</span> <span class="st">&quot;all&quot;</span>)</code></pre></div>
<pre><code>## =================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Smoking Score Calculation Started &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## =================================</code></pre>
<pre><code>## Dataset has 187 of 187 CpGs required for smoking status estimation.</code></pre>
<pre><code>## ====================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Smoking Score Calculation Completed &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ====================================</code></pre>
<pre><code>## ================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Methylation Score Calculation Started &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## =================================</code></pre>
<pre><code>## Dataset has 4 of 4 CpGs required for Zhang et al method</code></pre>
<pre><code>## ==================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Methylation Score Calculation Completed &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ==================================</code></pre>
<pre><code>## ============================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Predicting Smoking Statuses &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ============================================</code></pre>
<pre><code>## Dataset has 121 of 121 CpGs required for smoking status estimation.
## Dataset has 121 of 121 CpGs required for smoking status estimation.
## Dataset has 121 of 121 CpGs required for smoking status estimation.</code></pre>
<pre><code>## ============================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Prediction of Smoking Statuses Completed &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ============================================</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result_All</code></pre></div>
<pre><code>##                     SampleName smokingScore methylationScore logOdds_CS
## 1 GSM1051533_7800246024_R03C02    7.5861170        -4.284498  1.2702463
## 2 GSM1051534_7800246024_R04C02    1.1805043        -8.993818 -1.2369648
## 3 GSM1051538_7800246085_R02C01    1.1677897        -6.732453 -0.7931685
## 4 GSM1051539_7800246085_R03C01   -2.0578117        -7.180410 -1.2175070
## 5 GSM1051584_7800246137_R01C01   -0.2134197        -7.068201 -0.7364596
## 6 GSM1051585_7800246137_R02C01   -4.7081811        -8.731180 -1.5291249
##   logOdds_FS logOdds_NS   probs_CS  probs_FS   probs_NS
## 1  0.7726038 -2.0428501 0.60813729 0.3697245 0.02213819
## 2  0.2631190  0.9738458 0.06846889 0.3068820 0.62464911
## 3  0.3565658  0.4366027 0.13196440 0.4166583 0.45137726
## 4  0.4712640  0.7462431 0.07386166 0.3998002 0.52633818
## 5  0.5094571  0.2270025 0.14090711 0.4898100 0.36928289
## 6  0.6709051  0.8582198 0.04782438 0.4316286 0.52054702
##   PredictedSmokingStatus
## 1         Current Smoker
## 2           Never Smoker
## 3           Never Smoker
## 4           Never Smoker
## 5          Former Smoker
## 6           Never Smoker</code></pre>
</div>
</div>
<div id="b.-from-methylation-matrix" class="section level2">
<h2>B. From methylation matrix</h2>
<p>A data matrix of normalised methylation values in beta scale, with rows labelling the CpG probe ids and columns labelling the sample names, can also be used as an input to EpiSmokEr. Here we use the methylation matrix from the same 6 samples to demonstrate how methylation matrix can be used as input.</p>
<p>Methylation matrix looks likes this. Methylation values are in beta scale ranging betweeen 0 and 1.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(dummyBetaData)</code></pre></div>
<pre><code>##            GSM1051533_7800246024_R03C02 GSM1051534_7800246024_R04C02
## cg00006626                    0.8646877                    0.8431116
## cg00090764                    0.8318121                    0.8184256
## cg00997327                    0.8135604                    0.8176836
## cg01241450                    0.7683718                    0.8031366
## cg01926051                    0.1168564                    0.1108615
## cg02071454                    0.6653135                    0.6412269
##            GSM1051538_7800246085_R02C01 GSM1051539_7800246085_R03C01
## cg00006626                    0.8388067                    0.8478162
## cg00090764                    0.8198229                    0.8090982
## cg00997327                    0.8233350                    0.8225350
## cg01241450                    0.8074259                    0.7519214
## cg01926051                    0.1031007                    0.1329933
## cg02071454                    0.6219110                    0.6444712
##            GSM1051584_7800246137_R01C01 GSM1051585_7800246137_R02C01
## cg00006626                    0.8037606                    0.8561299
## cg00090764                    0.8077779                    0.8362229
## cg00997327                    0.7695912                    0.8095363
## cg01241450                    0.6968788                    0.7905101
## cg01926051                    0.1035927                    0.1106068
## cg02071454                    0.5116046                    0.6288115</code></pre>
<p>Samplesheet must contain the column <strong>sex</strong>, in the format of 1 and 2 representing men and women respectively.</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Age</th>
<th align="left">Gender</th>
<th align="left">Smokingstatus.selfReported.</th>
<th align="right">sex</th>
<th align="right">Slide</th>
<th align="left">Array</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GSM1051533_7800246024_R03C02</td>
<td align="right">55</td>
<td align="left">m</td>
<td align="left">current</td>
<td align="right">1</td>
<td align="right">7800246024</td>
<td align="left">R03C02</td>
</tr>
<tr class="even">
<td>GSM1051534_7800246024_R04C02</td>
<td align="right">37</td>
<td align="left">m</td>
<td align="left">never</td>
<td align="right">1</td>
<td align="right">7800246024</td>
<td align="left">R04C02</td>
</tr>
<tr class="odd">
<td>GSM1051538_7800246085_R02C01</td>
<td align="right">47</td>
<td align="left">m</td>
<td align="left">ex</td>
<td align="right">1</td>
<td align="right">7800246085</td>
<td align="left">R02C01</td>
</tr>
<tr class="even">
<td>GSM1051539_7800246085_R03C01</td>
<td align="right">69</td>
<td align="left">m</td>
<td align="left">never</td>
<td align="right">1</td>
<td align="right">7800246085</td>
<td align="left">R03C01</td>
</tr>
<tr class="odd">
<td>GSM1051584_7800246137_R01C01</td>
<td align="right">62</td>
<td align="left">f</td>
<td align="left">ex</td>
<td align="right">2</td>
<td align="right">7800246137</td>
<td align="left">R01C01</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">result &lt;-<span class="st"> </span><span class="kw">epismoker</span>(<span class="dt">dataset=</span>dummyBetaData, <span class="dt">samplesheet =</span> samplesheet, <span class="dt">method =</span> <span class="st">&quot;SSt&quot;</span>)</code></pre></div>
<pre><code>## ============================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Predicting Smoking Statuses &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ============================================</code></pre>
<pre><code>## Dataset has 121 of 121 CpGs required for smoking status estimation.
## Dataset has 121 of 121 CpGs required for smoking status estimation.
## Dataset has 121 of 121 CpGs required for smoking status estimation.</code></pre>
<pre><code>## ============================================</code></pre>
<pre><code>## &lt;&lt;&lt;&lt;&lt; Prediction of Smoking Statuses Completed &gt;&gt;&gt;&gt;&gt;</code></pre>
<pre><code>## ============================================</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">head</span>(result, <span class="dv">5</span>))</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="left">SampleName</th>
<th align="right">logOdds_CS</th>
<th align="right">logOdds_FS</th>
<th align="right">logOdds_NS</th>
<th align="right">probs_CS</th>
<th align="right">probs_FS</th>
<th align="right">probs_NS</th>
<th align="left">PredictedSmokingStatus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GSM1051533_7800246024_R03C02</td>
<td align="left">GSM1051533_7800246024_R03C02</td>
<td align="right">1.2702918</td>
<td align="right">0.7725532</td>
<td align="right">-2.0428449</td>
<td align="right">0.6081594</td>
<td align="right">0.3697025</td>
<td align="right">0.0221381</td>
<td align="left">Current Smoker</td>
</tr>
<tr class="even">
<td>GSM1051534_7800246024_R04C02</td>
<td align="left">GSM1051534_7800246024_R04C02</td>
<td align="right">-1.2368503</td>
<td align="right">0.2630556</td>
<td align="right">0.9737947</td>
<td align="right">0.0684797</td>
<td align="right">0.3068759</td>
<td align="right">0.6246444</td>
<td align="left">Never Smoker</td>
</tr>
<tr class="odd">
<td>GSM1051538_7800246085_R02C01</td>
<td align="left">GSM1051538_7800246085_R02C01</td>
<td align="right">-0.7930341</td>
<td align="right">0.3566457</td>
<td align="right">0.4363884</td>
<td align="right">0.1319882</td>
<td align="right">0.4167107</td>
<td align="right">0.4513012</td>
<td align="left">Never Smoker</td>
</tr>
<tr class="even">
<td>GSM1051539_7800246085_R03C01</td>
<td align="left">GSM1051539_7800246085_R03C01</td>
<td align="right">-1.2171854</td>
<td align="right">0.4711646</td>
<td align="right">0.7460208</td>
<td align="right">0.0738952</td>
<td align="right">0.3998136</td>
<td align="right">0.5262912</td>
<td align="left">Never Smoker</td>
</tr>
<tr class="odd">
<td>GSM1051584_7800246137_R01C01</td>
<td align="left">GSM1051584_7800246137_R01C01</td>
<td align="right">-0.7365582</td>
<td align="right">0.5093674</td>
<td align="right">0.2271908</td>
<td align="right">0.1408916</td>
<td align="right">0.4897603</td>
<td align="right">0.3693481</td>
<td align="left">Former Smoker</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="generate-results" class="section level1">
<h1>Generate Results</h1>
<hr />
<p>Results can be saved as html and csv files using <strong>generateReport</strong> function. A <strong>Results</strong> folder will be created in the <strong>current working directory</strong>. If <strong>Results</strong> folder already exists, result files will be appended to this folder.</p>
<p><strong>generateReport</strong> takes <strong>result object (resObj)</strong> and <strong>method</strong> used in epismoker function. Users have to provide a name for the output html file.</p>
<div id="please-note-that" class="section level4">
<h4>Please note that</h4>
<ul>
<li><p>In the R environment, result object from <strong>epismoker</strong> function must be saved using variable names <strong>result_SSt</strong>, <strong>result_SSc</strong>, <strong>result_MS</strong> and <strong>result_All</strong> based on the option used. This enables generation of html report.</p></li>
<li><p>In order to create the <strong>Results</strong> folder, set the working directory path to the folder where you have permission to write.</p></li>
</ul>
<p>In the example below result object from the <strong>all</strong> option of <strong>epismoker</strong> function has been saved to <strong>result</strong> variable, which will be passed onto generateReport function. On executing <strong>generateReport</strong> function, a <strong>Results</strong> folder will be created in the current working directory, with a html file named “Results_comprehensive.html”. Result files generated from the example below can be found in the <strong>Results</strong> folder of the package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">setwd</span>(<span class="kw">system.file</span>(<span class="dt">package =</span> <span class="st">&quot;EpiSmokEr&quot;</span>)) <span class="co"># set the directory where you want the result files to be saved</span>
<span class="kw">generateReport</span>(<span class="dt">resObj =</span> result_All, <span class="dt">outputFileName =</span> <span class="st">&quot;Results_comprehensive&quot;</span>, <span class="dt">method =</span> <span class="st">&quot;all&quot;</span>)</code></pre></div>
<pre><code>## ====================================================================================================================</code></pre>
<pre><code>## Saving results as html and csv files to C:\Users\sairam\Documents\R\win-library\3.5\Results</code></pre>
<pre><code>## ====================================================================================================================</code></pre>
<pre><code>## 
## 
## processing file: all.rmd</code></pre>
<pre><code>## 
  |                                                                       
  |                                                                 |   0%
  |                                                                       
  |...........                                                      |  17%
##    inline R code fragments
## 
## 
  |                                                                       
  |......................                                           |  33%
## label: unnamed-chunk-20 (with options) 
## List of 2
##  $ echo   : logi FALSE
##  $ results: chr &quot;asis&quot;
## 
## 
  |                                                                       
  |................................                                 |  50%
##   ordinary text without R code
## 
## 
  |                                                                       
  |...........................................                      |  67%
## label: unnamed-chunk-21 (with options) 
## List of 2
##  $ echo   : logi FALSE
##  $ results: chr &quot;asis&quot;
## 
## 
  |                                                                       
  |......................................................           |  83%
##   ordinary text without R code
## 
## 
  |                                                                       
  |.................................................................| 100%
## label: unnamed-chunk-22 (with options) 
## List of 1
##  $ results: chr &quot;markup&quot;</code></pre>
<pre><code>## output file: all.knit.md</code></pre>
<pre><code>## &quot;C:/Program Files/RStudio/bin/pandoc/pandoc&quot; +RTS -K512m -RTS all.utf8.md --to html4 --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output pandoc2ff01c73588e.html --smart --email-obfuscation none --self-contained --standalone --section-divs --template &quot;C:\Users\sairam\Documents\R\win-library\3.5\rmarkdown\rmd\h\default.html&quot; --no-highlight --variable highlightjs=1 --variable &quot;theme:bootstrap&quot; --include-in-header &quot;C:\Users\sairam\AppData\Local\Temp\RtmpmcwAcO\rmarkdown-str2ff04dcc389b.html&quot; --mathjax --variable &quot;mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML&quot;</code></pre>
<pre><code>## 
## Output created: C:/Users/sairam/Documents/R/win-library/3.5/Results/Results_comprehensive.html</code></pre>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<hr />
<div id="refs" class="references">
<div id="ref-minfi">
<p>Aryee, Martin J, Andrew E Jaffe, Hector Corrada-Bravo, Christine Ladd-Acosta, Andrew P Feinberg, Kasper D Hansen, and Rafael A Irizarry. 2014. “Minfi: A Flexible and Comprehensive Bioconductor Package for the Analysis of Infinium Dna Methylation Microarrays.” <em>Bioinformatics</em> 30 (10): 1363–9. doi:<a href="https://doi.org/10.1093/bioinformatics/btu049">10.1093/bioinformatics/btu049</a>.</p>
</div>
<div id="ref-elliott">
<p>Elliott, Hannah R, Therese Tillin, Wendy L McArdle, Karen Ho, Aparna Duggirala, Tim M Frayling, George Davey Smith, Alun D Hughes, Nish Chaturvedi, and Caroline L Relton. 2014. “Differences in Smoking Associated Dna Methylation Patterns in South Asians and Europeans.” <em>Clinical Epigenetics</em> 6 (1). doi:<a href="https://doi.org/10.1186/1868-7083-6-4">10.1186/1868-7083-6-4</a>.</p>
</div>
<div id="ref-geo">
<p>Liu, Yun, Martin J Aryee, Leonid Padyukov, Daniele M Fallin, Espen Hesselberg, Arni Runarsson, Lovisa Reinius, et al. 2013. “Epigenome-Wide Association Data Implicate Dna Methylation as an Intermediary of Genetic Risk in Rheumatoid Arthritis.” <em>Nature Biotechnology</em> 31: 142–47. doi:<a href="https://doi.org/doi:10.1038/nbt.2487">doi:10.1038/nbt.2487</a>.</p>
</div>
<div id="ref-tost">
<p>Touleimat, Nizar, and Jörg Tost. 2012. “Complete Pipeline for Infinium® Human Methylation 450k Beadchip Data Processing Using Subset Quantile Normalization for Accurate Dna Methylation Estimation.” <em>Epigenomics</em> 4 (3): 325–41. doi:<a href="https://doi.org/10.2217/epi.12.21">10.2217/epi.12.21</a>.</p>
</div>
<div id="ref-zellinger">
<p>Zeilinger, Sonja, Brigitte Kühnel, Norman Klopp, Hansjörg Baurecht, Anja Kleinschmidt, Christian Gieger, Stephan Weidinger, et al. 2013. “Tobacco Smoking Leads to Extensive Genome-Wide Changes in Dna Methylation.” <em>PLOS ONE</em> 8 (5). Public Library of Science: 1–14. doi:<a href="https://doi.org/10.1371/journal.pone.0063812">10.1371/journal.pone.0063812</a>.</p>
</div>
<div id="ref-zhang">
<p>Zhang, Yan, Ines Florath, Kai-Uwe Saum, and Hermann Brenner. 2016. “Self-Reported Smoking, Serum Cotinine, and Blood Dna Methylation.” <em>Environmental Research</em> 146. doi:<a href="https://doi.org/10.1016/j.envres.2016.01.026">10.1016/j.envres.2016.01.026</a>.</p>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
